src/Common/src/System/Net/Http/TlsCertificateExtensions.cs

```diff
@@ -73,6 +73,24 @@ private static bool IsClientCertificate(X509Certificate2 cert)
             return isClientAuth && isDigitalSignature;
         }
 
+        internal static X509Chain BuildNewChain(X509Certificate2 certificate)
+        {
+            var chain = new X509Chain();
+            chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+            chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+            chain.ChainPolicy.ApplicationPolicy.Add(s_clientCertOidInst);
```

Comment:
This ApplicationPolicy element shouldn't be set for the single-cert case; it'll cause the chain to fail to build if the cert isn't an appropriate client auth cert, but we're going to select the leaf cert anyways.